network/src/worker.rs

```diff
@@ -27,32 +29,33 @@ impl Default for WorkerNetwork {
 
         Self {
             clients: Default::default(),
+            config: Default::default(),
             retry_config,
             rng: SmallRng::from_entropy(),
         }
     }
 }
 
 impl WorkerNetwork {
-    fn client(&mut self, address: SocketAddr) -> WorkerToWorkerClient<Channel> {
+    fn client(&mut self, address: Multiaddr) -> WorkerToWorkerClient<Channel> {
         self.clients
-            .entry(address)
-            .or_insert_with(|| Self::create_client(address))
+            .entry(address.clone())
+            .or_insert_with(|| Self::create_client(&self.config, address))
             .clone()
     }
 
-    fn create_client(address: SocketAddr) -> WorkerToWorkerClient<Channel> {
-        // TODO use TLS
```

Comment:
And ditto: I'd like to not lose this comment, because in the mTLS context, using TLS is not just about a change of protocol, but requires configuring a custom verifier.